terraform/environments/core-network-services/iam.tf

```diff
@@ -16,6 +16,17 @@ resource "aws_iam_role" "dns" {
           },
           "Action" : "sts:AssumeRole",
           "Condition" : {}
+          }, {
+          "Effect" : "Allow",
+          "Principal" : {
+            "AWS" : "arn:aws:iam::*:role/github-actions"
+          },
+          "Action" : "sts:AssumeRole",
+          "Condition" : {
+            "ForAnyValue:StringLike" : {
+              "aws:PrincipalOrgPaths" : ["${data.aws_organizations_organization.root_account.id}/*/${local.environment_management.modernisation_platform_organisation_unit_id}/*"]
```

Comment:
Looks good, I think we can lock this down further to the member ou, but we don't have that stored as a secret, so I think it's best to go with this for now and then lock it down further as part of that other issue.